pkg/asset/asset.go

```diff
@@ -81,3 +86,105 @@ func AncestryPath(ancestors []string) string {
 	}
 	return strings.Join(revAncestors, "/")
 }
+
+// UnwrapCAIResource will unwrap a K8S resource from the CAI payload and populate any omitted fields.
+func UnwrapCAIResource(asset map[string]interface{}) (*unstructured.Unstructured, error) {
+	groupKind, found, err := unstructured.NestedString(asset, "asset_type")
+	if !found {
+		return nil, errors.Errorf("asset_type field not found")
+	}
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to access asset_type field")
+	}
+
+	parts := strings.Split(groupKind, "/")
+	if len(parts) != 2 {
+		return nil, errors.Errorf("expected asset_type to be of form \"<group>/<kind>\", got %s", groupKind)
+	}
+	group := parts[0]
+	kind := parts[1]
+	if group == "k8s.io" {
```

Comment:
maybe add a comment to explain why we need to erase it?

Reply:
added